src/main/include/subsystems/SwerveDrive.h

```diff
@@ -91,34 +87,16 @@ class SwerveDrive : public frc2::SubsystemBase {
     std::shared_ptr<SwerveModule> right_front_module_;
     std::shared_ptr<SwerveModule> right_back_module_;
 
-    // swerve module positions
-    std::shared_ptr<frc::Translation2d> left_front_loc_;
-    std::shared_ptr<frc::Translation2d> left_back_loc_;
-    std::shared_ptr<frc::Translation2d> right_front_loc_;
-    std::shared_ptr<frc::Translation2d> right_back_loc_;
-
-    // swerve module states
-    std::shared_ptr<frc::SwerveModulePosition> left_front_pos_;
-    std::shared_ptr<frc::SwerveModulePosition> left_back_pos_;
-    std::shared_ptr<frc::SwerveModulePosition> right_front_pos_;
-    std::shared_ptr<frc::SwerveModulePosition> right_back_pos_;    
-
-    // swerve module positions
-    std::shared_ptr<wpi::array<frc::SwerveModulePosition, 4>> positions_;
-
-    // kinematics
-    std::shared_ptr<frc::SwerveDriveKinematics<4>> swerve_kinematics_;
-
-    // odometry
-    std::shared_ptr<frc::SwerveDriveOdometry<4>> swerve_odometry_;
-
-    // position
-    std::shared_ptr<frc::Pose2d> position_;
-
     // Speed modifier (the joystick input is multiplied by this value)
     double speed_modifier_drive_ = 0.75;
     double speed_modifier_steer_ = 0.75;
 
+    double last_drive = 0.0;
+    double last_strafe = 0.0;
+    double last_turn = 0.0;
+
+    double control_decay = 0.1;
+
```

Comment:
_

Reply:
yeah @noahzemlin said you wouldn't like it. I can parameterize it though.

Reply:
#45